src/Microsoft.AspNet.Http.Abstractions/Authentication/AuthenticationManager.cs

```diff
@@ -11,9 +12,24 @@ public abstract class AuthenticationManager
     {
         public abstract IEnumerable<AuthenticationDescription> GetAuthenticationSchemes();
 
-        public abstract AuthenticationResult Authenticate(string authenticationScheme);
+        public abstract void Authenticate(AuthenticateContext context);
 
-        public abstract Task<AuthenticationResult> AuthenticateAsync(string authenticationScheme);
+        public abstract Task AuthenticateAsync(AuthenticateContext context);
```

Comment:
That is fairly uncommon which is why we are leaning this way to start (Identity will be using the properties, so when I get there, if its super bad, don't worry I'll definitely add sugar :)

But most of the time, Authenticate() giving you back a claimsprincipal really all you want, without having to dig through the old Result tuple...